File: ruby/pigatize.rb

def pigatize(text)
  if starts_with_vowel?(text)
    puts text + "way"
  else
    new_text = text.chars
    piggie = new_text.shift
    piggie.downcase!
    puts new_text.join('') + piggie + "ay"
  end
end

def starts_with_vowel?(text)
  if /^[aeiouAEIOU]/.match(text)
    return true
  end
end

pigatize("Aardvark")
pigatize("Bob")
pigatize("Reginald")

File: javascript/pigatize.js

var pigatize = function(text) {
    if (isAVowel(text) === true) {
        console.log(text + "way");
    }
    else {
        var first = text[0];
        var lowerFirst = first.toLowerCase();
        console.log(text.slice(1) + lowerFirst + "ay");
    }
};

var isAVowel = function(text) {
    if (/^[aeiouAEIOU]/.test(text))
        return true;
    else {
        return false;
    }
};

pigatize("George");
pigatize("Aardvark");
pigatize("Robert");
pigatize("Orange");
